### ISSUE TITLE
Introduce Lance as a new basic format for lerobot

### ISSUE THREAD

**Issue by yanghua** [huggingface/lerobot#2417]
Hi forks,

I'd like to kick off an idea about introducing Lance[1] as a new basic format.

Currently, there are three types of file formats that compose a Lerobot generic dataset(correct me if I am wrong). They are:

* parquet: stores structured features;
* mp4: stores videos;
* json/jsonl: stores meta information;

I am thinking if we can add Lance as a new supported (file) format for Lerobot, or maybe we can merge some of the above formats into the Lance (table) format. Let me explain my thoughts.

In short, Lance is a modern multi-modal columnar format that is becoming more and more popular in managing and training multi-modal data. It combines file format(compared to Parquet) and table format(compared to Iceberg) into one. And providing some benefits we can get from it:

* **Easy to manage(All data in one format and one table)**: Lance supports structured, semi-structured, and non-structured data(binary and blob) into one two-dimensional table. That means we can store episodes' videos, features/labels, and sensors(nested structure) in one row per episode;
* **Fast random access and scan**: it's essential for ML training. Lance supports row-id as its first-level index, and supports scalar, FTS, and vector indices as its second-level index. We can shuffle all the episodes via row index, row ID, or just filter with a condition in a PyTorch Dataloader when training.
* **Zero-cost data evolution**: it's very lightweight(compared to Parquet) for adding new feature/label(columns) and back-fill features/labels for episode.
* **Data exploration-friendly**: it natively supports SQL API and hybrid search(vector and scalar prefilter), what's more, DuckDB is also in the supported scope[5].
*  **AI-ecosystem-friendly**: Lance can be easily integrated with Huggingface[2], Pytorch[3] and Ray[4], Daft for high-performance processing and training.

Generally speaking, there are many benefits we can get from Lance(still many features are not listed, lol). No matter whether we support it at the file format level(Lance File API) or the table format level(Lance Dataset API).

Any inputs or feedback are welcome.

[1]: https://github.com/lancedb/lance
[2]: https://lancedb.github.io/lance/integrations/huggingface/
[3]: https://lancedb.github.io/lance/integrations/pytorch/
[4]: https://lancedb.github.io/lance/integrations/ray/
[5]: https://lancedb.github.io/lance/integrations/duckdb/

**Comment by yanghua** [comment]
Hi @jadechoghari, WDYT about this proposal?

**Comment by yanghua** [comment]
Or maybe @imstevenpmwork can share some thoughts. Sorry, I am a newbie in this project, not sure if there is someone who mainly maintains the dataset module. The reason I raise this proposal is that we build an AI multi-modal data lake solution around Lance. And some customers use Lerobot, we found that combining these two projects is a win-win.

**Comment by jadechoghari** [comment]
Hey @yanghua ; that could be considered for LeRobotDataset v4 ; however note that parquet is tightly coupled with huggingface [datasets](https://huggingface.co/docs/datasets/en/index) which LeRobot uses under the hood

**Comment by yanghua** [comment]
Hi @jadechoghari, Thanks for your reply!

> however note that parquet is tightly coupled with huggingface [datasets](https://huggingface.co/docs/datasets/en/index) which LeRobot uses under the hood

I quickly checked the source code to see how Lerobot downloads the HuggingFace dataset; it uses the `datasets.load_dataset` function to achieve this(correct me if I am wrong).

https://github.com/huggingface/lerobot/blob/a5b29d430105f5235eb05bbf2db5a0d747a869d6/src/lerobot/datasets/streaming_dataset.py#L152

On the other side, Lance can also integrate with Huggineface with a similar style, like this:

```python
import datasets # pip install datasets
import lance

lance.write_dataset(datasets.load_dataset(
    "poloclub/diffusiondb", split="train[:10]",
), "diffusiondb_train.lance")
```

In short, IMO, the cost seems very low, right?

**Comment by yanghua** [comment]
I am trying to open a PR to support this feature. cc @jadechoghari

**Comment by zhang1002** [comment]
when i designed the metadata management service for my company, the embodied intelligence, i thought if i could use lance to manage the data...

**Comment by yanghua** [comment]
> when i designed the metadata management service for my company, the embodied intelligence, i thought if i could use lance to manage the data...

Hi @zhang1002, Thanks for your feedback. It means that this feature is really valuable. Let's make it happen!

**Comment by eddyxu** [comment]
@jadechoghari

Huggingface datasets now support lance format, 

- https://huggingface.co/docs/hub/datasets-lance
- https://www.lancedb.com/blog/lance-x-huggingface-a-new-era-of-sharing-multimodal-data/

is there anything we can help to make this happen?

**Comment by lhoestq** [comment]
Lance released their own LeRobotDataset implementation :) see https://github.com/huggingface/lerobot/issues/3608
